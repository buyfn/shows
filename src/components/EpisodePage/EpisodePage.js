import React from 'react';

class EpisodePage extends React.Component {
  componentDidMount() {
    const episodeID = this.props.match.params.episodeID;
    this.props.fetchEpisode(episodeID);
  }

  render() {
    const { image, name, summary, fetchingState, errorMsg } = this.props;

    if (fetchingState === 'fetching') {
      return 'Fetching data...';
    }

    if (fetchingState === 'failed') {
      return errorMsg;
    }

    return (
      <>
        {image && (
          <img alt={`"${name}" episode cover`} src={image.original} />
        )}

        <h1>{name}</h1>

        <div dangerouslySetInnerHTML={{ __html: summary }} />
      </>
    );
  }
}

export default EpisodePage;
