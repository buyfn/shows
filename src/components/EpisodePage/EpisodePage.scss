.episode-page {
    max-width: 1400px;
    margin: 0 2em;
}

.episode-image {
    max-width: 100%;
}

.episode-info {
    max-width: 800px;
}