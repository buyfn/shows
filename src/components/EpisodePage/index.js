import { connect } from 'react-redux';

import { fetchEpisodeRequest } from '../../actions';

import EpisodePage from './EpisodePage';

const mapStateToProps = (state) => ({
  ...state.episode,
});

const mapDispatchToProps = dispatch => ({
  fetchEpisode: (id) => dispatch(fetchEpisodeRequest(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(EpisodePage);
