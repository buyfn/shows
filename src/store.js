import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

import reducer from './reducers';
import { saga } from './sagas';

const sagaMiddleware = createSagaMiddleware();

const composeEnchancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducer,
  composeEnchancers(
    applyMiddleware(sagaMiddleware),
  ),
);

sagaMiddleware.run(saga);

export default store;
