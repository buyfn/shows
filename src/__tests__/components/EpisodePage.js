import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../../store';

import EpisodePage from '../../components/EpisodePage/EpisodePage';

describe('EpisodePage', () => {
  const props = {
    match: {
      params: { episodeID: '1' }
    },
    summary: 'test episode summary',
    name: 'test episode name',
    image: {
      medium: 'img.jpg',
      original: 'img.jpg'
    },
  };

  it('renders correctlly', () => {
    const tree = renderer
      .create(
        <Provider store={store}><EpisodePage {...props}/></Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
