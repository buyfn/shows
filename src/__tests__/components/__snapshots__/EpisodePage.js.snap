// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePage renders correctlly 1`] = `
<main
  className="episode-page"
>
  <div
    className="episode-info"
  >
    <h1
      className="episode-title"
    >
      test episode name
    </h1>
    <div
      className="episode-description"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test episode summary",
        }
      }
    />
  </div>
  <img
    alt="\\"test episode name\\" episode cover"
    className="episode-image"
    src="img.jpg"
  />
</main>
`;
